app/imports/components/fields/UploadField.jsx

```diff
@@ -0,0 +1,77 @@
+/* globals Slingshot */
+import React, { Component } from 'react';
+
+import { connectField } from 'uniforms';
+
+import '/imports/startup/slingshot.js';
+
+class FieldUpload extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      uploading: false,
+    };
+  }
+
+  uploadFile() {
+    const image = this.refs.fileInput.files[0];
+    const uploader = new Slingshot.Upload('imageUpload');
+
+    this.setState({
```

Comment:
Yes saw below. Does it also do events or just that state inside the declaration stuff?